bin/pycbc_optimize_snr

```diff
@@ -142,12 +157,13 @@ def optimize_di(bounds, cli_args, extra_args):
         mutation=(0.5, 1),
         recombination=0.7,
         callback=callback_func,
-        args=extra_args
+        args=extra_args,
+        init=population
     )
     return results.x
 
 
-def optimize_shgo(bounds, cli_args, extra_args):
+def optimize_shgo(bounds, cli_args, extra_args, initial_point): # pylint: disable=unused-argument
```

Comment:
IMO the `shgo` optimizer could be removed, however that would not be the point of this PR and I think a dedicated PR with specific reasoning and clear objectives would be best to do that - thanks for volunteering to do that!

the #pylint thing is as the `initial_point` variable is not used in the function, so it would be seen as an error by the CI checks, but we want to be able to pass it for the helper function

Reply:
Ah okay, that makes sense! Yes, I think removing the optimizer should be considered to clean this code up as I think it's gotten quite messy.